### PR TITLE
Terminal and Commands

### PR DESCRIPTION
**Pull request by Loerad**
This PR adds the Terminal Window and Commands

The terminal is activated with F1 and you can use "help" to find available commands from there. 

This should help us going further test things and also break things if we wish.

Current commands:
```
"setweapon {weaponName}     | will set player weapon to weapon given. Accepted values: pistol, dualpistol, shotgun, sniper, machinegun, rocket, sword" 
"giveitem {itemID} {amount} | will give an item or multiple from the item list to the player (will not add to list of items the player has)"
"setstat {stat} {value}     | sets the stat specified to the value given. Accepted stats: damage, crit, maxhealth, firedelay, movespeed, critdamage"
"setwave {waveNo.}          | sets the next wave to the specified number"
"skipwave                   | will set the timer to 0, skipping the wave you are on"
"godmode {bool}             | gives the player god mode. true/false"
"spawn {enemyId} {count}    | spawns an enemy with the id given and the amount given"
"cull                       | will cull all current enemies and bullets on screen"
"stopspawn {value}          | will toggle enemies or bosses from spawning. Accepted values: enemy, boss"
```
If any command seems confusing or not explained enough feel free to write a suggestion

### CODE REVIEW

**Comment by Loerad** [comment]
PR is ready to review

**Review by AlexReidNZ — CHANGES_REQUESTED** [review]
All the current commands work as they are intended, togglespawn needs its name updated everywhere that isn't the help menu then will be ready for merging.

File: Assets/Scripts/Terminal/TerminalBehaviour.cs [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+                    output.text += "\ncull\n\nCulls all active enemies and bullets on stage.\n\n";
+                break;
+                case "togglespawn":
+                    output.text += "\nstopspawn {value:string}\n\nWill stop either enemies or bosses, depeding on value given, from spawning further. Does not cull current eneimes, ues \"cull\" to do that.\n\nAccepted values: \nenemy, \nboss \n\n";
+                break;
+                case "skipwave":
+                    output.text += "\nskipwave\n\nSkips the current wave by setting the timer to 0.1.\n\n";
+                break;
+                case "spawn":
+                    output.text += "\nspawn {enemyID:int} {count:int}\n\nSpawns the amount of eneimes specified in count, and the type of enemy in enemyID.\n\n";
```
Comment (PBWiseman) [inline Assets/Scripts/Terminal/TerminalBehaviour.cs:158]: Enemies instead of eneimes

File: Assets/Scripts/Terminal/TerminalBehaviour.cs [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+                    output.text += "\nsetstat {waveNo.:int}\n\nWill set the next wave to the number given.\n\n";
+                break;
+                case "godmode":
+                    output.text += "\ngodmode\n\nWill toggle between turning on and off the players ability to be damaged, will not heal the player.\n\n";
+                break;
+                case "cull":
+                    output.text += "\ncull\n\nCulls all active enemies and bullets on stage.\n\n";
+                break;
+                case "togglespawn":
+                    output.text += "\nstopspawn {value:string}\n\nWill stop either enemies or bosses, depeding on value given, from spawning further. Does not cull current eneimes, ues \"cull\" to do that.\n\nAccepted values: \nenemy, \nboss \n\n";
```
Comment (PBWiseman) [inline Assets/Scripts/Terminal/TerminalBehaviour.cs:152]: Use instead of ues, enemies instead of eneimes, depending instead of depeding

**Review by PBWiseman — CHANGES_REQUESTED** [review]
Terminal works great! Will make it much easier to test features. The only major issue I could find is that if you input an enemy to spawn that doesn't exist (anything over 2 at the moment) it just has an error and doesn't print anything. Also a couple of small spelling errors to fix in review comments

File: Assets/Scripts/Terminal/TerminalBehaviour.cs [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+    }
+
+    private void Cull()
+    {
+        Timer.CullBullets();
+        Timer.CullEnemies();
+        output.text += "\nAll enemies on screen culled\n\n";
+    }
+
+    private void StopSpawn(string choice)
```
[the remark this answers is not in the document]
Comment (Loerad) [reply Assets/Scripts/Terminal/TerminalBehaviour.cs:311]: fixed

**Review by PBWiseman — APPROVED** [review]
All issues addressed. Good to merge

**Review by AlexReidNZ — APPROVED** [review]
Requested changes made, good for merge.

**Pull request merged**